src/interchange/toml/lexer.zig

```diff
@@ -586,41 +584,15 @@ pub const Lexer = struct {
                     lexer.token = T.t_comma;
                 },
                 ';' => {
-                    if (lexer.has_newline_before) {
-                        lexer.step();
-
-                        singleLineComment: while (true) {
-                            lexer.step();
-                            switch (lexer.code_point) {
-                                '\r', '\n', 0x2028, 0x2029 => {
-                                    break :singleLineComment;
-                                },
-                                -1 => {
-                                    break :singleLineComment;
-                                },
-                                else => {},
-                            }
-                        }
-                        continue;
+                    if (!lexer.has_newline_before) {
+                        try lexer.addDefaultError("Unexpected semicolon");
                     }
 
-                    try lexer.addDefaultError("Unexpected semicolon");
+                    lexer.discardTilEndOfLine();
+                    continue;
                 },
                 '#' => {
-                    lexer.step();
-
-                    singleLineComment: while (true) {
-                        lexer.step();
-                        switch (lexer.code_point) {
-                            '\r', '\n', 0x2028, 0x2029 => {
-                                break :singleLineComment;
-                            },
-                            -1 => {
-                                break :singleLineComment;
-                            },
-                            else => {},
-                        }
-                    }
+                    lexer.discardTilEndOfLine();
```

Comment:
Seems like a correct change. The old implementation seems like AI.

Reply:
💯